include/oneapi/mkl.hpp

```diff
@@ -26,5 +26,6 @@
 #include "oneapi/mkl/dft.hpp"
 #include "oneapi/mkl/lapack.hpp"
 #include "oneapi/mkl/rng.hpp"
+#include "oneapi/mkl/sparse_blas.hpp"
```

Comment:
Please rename this file to `spblas.hpp` to align with the oneMKL product.
```suggestion
#include "oneapi/mkl/spblas.hpp"
```

Reply:
So I tried to do this change but this is more complicated than it looks. We run into the issue that the backend is not able to include the close-source headers anymore. The issue is that the CMake target includes the close-source MKL headers as system headers (from the target `MKL::MKL_DPCPP`) which forces the compiler to search for the close-source headers last. If the oneMKL interface project has both the files `mkl.hpp` and `spblas.hpp` then I can't find a way to use the MKL C++ api.
From what I understand for the other domains the situation varies a bit. The blas and lapack domain seem to never include the close-source headers, they redefine the same types and prototypes in this project. I don't think that would be as maintainable for sparse blas since the API is not as stable.
The rng domain seem to use the C API (headers have different name so this works fine)
For DFT, the MKLCPU backend uses the C API and the MKLGPU backend uses the C++ API but the headers have different names (`dft.hpp` in oneMKL interface and `dfti.hpp` in oneMKL product).
Are you sure it is required that each domain's headers match between the interface and the product? I think this can cause issues for the users as well, this is one of the issue that I described in https://github.com/oneapi-src/oneMKL/issues/306. If it is important then I can switch to using the C API for MKLCPU (and MKLGPU backend in the future).

Reply:
@mkrainiuk I would be curious to hear your opinion on this. TL;DL is that I'm having issues trying to get this working if we rename `oneapi/mkl/sparse_blas.hpp` -> `oneapi/mkl/spblas.hpp`. Maybe you know whether the header names must match with the ones in the oneapi package?

Reply:
@mkrainiuk, @Rbiessy, @spencerpatty, what are your thoughts on this? I thought the names should be aligned because this repo started off as Intel's implementation of the oneMKL Spec (and still is?), and we would have preferred to keep the header names same if possible. For that matter, I'm not sure if the spec even talks about what the header names should be, or if it even should specify header names at all. It makes sense to leave header names for implementations to decide on, in which case, having `sparse_blas.hpp` or any other header name is completely fine. Nevertheless, for this particular repository, do we have a requirement to keep the header names same between this repo (oneMKL Interfaces) and Intel oneMKL?

Reply:
I understand that this is a complicated issue from a language perspective and making it the same name is not possible.  I am completely fine with having the `sparse_blas.hpp` name being used in the Open Source project declarations, so that in the backends, we can use a different header solution say, `spblas.hpp` for our close source headers...  

A separate solution (not saying it is best, but it is something to consider as this is a real issue for several domains, especially since our close source headers may have much more defined than is in the spec, there really should be separate definitions... ) would be to add something to the open source namespace implementation that would allow to distinguish between oneMKL Interfaces and oneMKL product ...  maybe something like the following ??

`oneapi::mkl::interfaces::spblas/blas/dft/etc`
or 
`oneapi::mkl_interfaces::spblas/blas/dft/etc`  ??

Reply:
@mkrainiuk and @Rbiessy and @gajanan-choudhary what do you think about something like that for long term solution ?  In short term, using sparse_blas.hpp is probably fine.

Reply:
So I would be in favor of having a different namespace for oneMKL interface as it would make it clearer which libraries are used. From what I understand this was done intentionally so users could transition from the close-source libraries to the open-source one more easily. I think the current situation can be confusing for new users.

However I don't think this would solve the issue with the same file names. This issue really boils down to the compiler rules for including files and the current CMake architecture, changing the namespace does not affect that. The MKL backends would still not be able to include the close-source headers. If you ask me, the proper solution would be to change the close-source CMake library `MKL::MKL_DPCPP` so that it does not include the headers as system headers. It may also be possible to work around it by changing the oneMKL interface CMake so that `ONEMKL::SYCL::SYCL` does not include some directories that are not needed for building `onemkl_sparse_blas_mklcpu`.

I'm happy with the solution of keeping the current file names. If we need to change this later I could spend some time to better explain the issue and come up with a small reproducer.

Reply:
Ah, yes I see your point that naemspaces wouldn't help here.  Yes, lets go ahead and keep as is for now and maybe explore for later what could be done.  I don't understand the CMake issue either, I would think that all inclusions of headers should be specialized (not treated as system headers), but I am sure there are things I am not seeing right now which lead to the current design...